components/profile/unclaimed-hypercert-claim-button.tsx

```diff
@@ -27,9 +29,20 @@ export default function UnclaimedHypercertClaimButton({
   const { setDialogStep, setSteps, setOpen, setTitle, setExtraContent } =
     useStepProcessDialogContext();
   const { switchChain } = useSwitchChain();
+  const queryClient = useQueryClient();
+  const router = useRouter();
+
   const selectedHypercert = allowListRecord.original;
   const hypercertChainId = selectedHypercert?.hypercert_id?.split("-")[0];
 
+  const refreshData = async (address: string) => {
+    await queryClient.invalidateQueries({
+      queryKey: ["hypercerts-data", address.toLowerCase()],
+    });
+    await revalidatePathServerAction(`/profile/${address}`);
+    router.refresh();
```

Comment:
Do we need to refresh? Seems to me that invalidating the queries should be enough to trigger a new fetch?

Reply:
Can't tell you this at the moment as the API isn't invalidating the data that it should. The refresh was already in the batch claim button and I just made them consistent.

Also, according to the docs invalidating dynamic paths doesn't apply without refetching: https://nextjs.org/docs/14/app/api-reference/functions/revalidatePath